src/wire/ipv6option.rs

```diff
@@ -0,0 +1,367 @@
+use core::fmt;
+use {Error, Result};
+
+enum_with_unknown! {
+    /// IPv6 Extension Header Option Type
+    pub doc enum Type(u8) {
+        /// 1 byte of padding
+        Pad1 =  0,
+        /// Multiple bytes of padding
+        PadN =  1
+    }
+}
+
+impl fmt::Display for Type {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        match self {
+            &Type::Pad1        => write!(f, "Pad1"),
+            &Type::PadN        => write!(f, "PadN"),
+            &Type::Unknown(id) => write!(f, "{}", id)
+        }
+    }
+}
+
+/// A read/write wrapper around an IPv6 Extension Header Option.
+#[derive(Debug, PartialEq)]
+pub struct Ipv6Option<T: AsRef<[u8]>> {
+    buffer: T
+}
+
+// Format of Option
+//
+// +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+- - - - - - - - -
+// |  Option Type  |  Opt Data Len |  Option Data
+// +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+- - - - - - - - -
+//
+//
+// See https://tools.ietf.org/html/rfc8200#section-4.2 for details.
+mod field {
+    use wire::field::*;
+
+    // 8-bit identifier of the type of option.
+    pub const TYPE:     usize = 0;
+    // 8-bit unsigned integer. Length of the DATA field of this option, in octets.
+    pub const LENGTH:   usize = 1;
+    // Variable-length field. Option-Type-specific data.
+    pub fn DATA(length: u8) -> Field {
```

Comment:
`allow(non_snake_case)` if we're sticking to `DATA`.

Reply:
i have this fixed in my latest changes

Reply:
:+1: Thanks! Just a side note. I personally I tend to prefer using

```rust
mod something {
    // other stuff in the module

    #[allow(non_snake_case)]
    fn EXCEPTIONTOTHERULE() {
    }
}
```

instead of

```rust
mod something {
    #![allow(non_snake_case)]

    // other stuff in the module

    fn EXCEPTIONTOTHERULE() {
    }
}
```

Reply:
No, this is fine in `mod fields`, since it has a specific structure.

Reply:
I was following the convention as I saw it here: https://github.com/m-labs/smoltcp/blob/c2d18ec071a2cd68aee2724ce13eefa6ccf1f0c1/src/wire/tcp.rs#L75